File: Tests.Common/Objects/CSCompiler/Unary.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Linq.Expressions;
using System.Text;
using System.Threading.Tasks;
using static ExpressionToString.Tests.Categories;
using static ExpressionToString.Tests.Functions;


namespace ExpressionToString.Tests.Objects {
    partial class CSCompiler {
        [Category(Unary)]
        public static readonly Expression Negate = IIFE(() => {
            var i = 1;
            return Expr(() => -i);
        });

        [Category(Unary)]
        public static readonly Expression BitwiseNot = IIFE(() => {
            var i = 1;
            return Expr(() => ~i);
        });

        [Category(Unary)]
        public static readonly Expression LogicalNot = IIFE(() => {
            var b = true;
            return Expr(() => !b);
        });

        [Category(Unary)]
        public static readonly Expression TypeAs = IIFE(() => {
            object o = null;
            return Expr(() => o as string);
        })
            ;
        [Category(Unary)]
        public static readonly Expression ArrayLength = IIFE(() => {
            var arr = new string[] { };
            return Expr(() => arr.Length);
        });

        [Category(Unary)]
        public static readonly Expression Convert = IIFE(() => {
            var lst = new List<string>();
            return Expr(() => (object)lst);
        });
    }
}

File: Tests.Common.VB/Objects/Unary.vb
Imports ExpressionToString.Tests.Functions

Namespace Objects
    Partial Public Module VBCompiler
        <Category(Unary)>
        Public ArrayLength As Expression = IIFE(Function()
                                                    Dim arr = New String() {}
                                                    Return Expr(Function() arr.Length)
                                                End Function)

        <Category(Unary)>
        Public Convert As Expression = IIFE(Function()
                                                Dim o As Object = New Random
                                                Return Expr(Function() CType(o, Random))
                                            End Function)

        <Category(Unary)>
        Public CObject As Expression = IIFE(Function()
                                                Dim lst = New List(Of String)()
                                                Return Expr(Function() CObj(lst))
                                            End Function)

        <Category(Unary)>
        Public Negate As Expression = IIFE(Function()
                                               Dim i = 1
                                               Return Expr(Function() -i)
                                           End Function)

        <Category(Unary)>
        Public BitwiseNot As Expression = IIFE(Function()
                                                   Dim i = 1
                                                   Return Expr(Function() Not i)
                                               End Function)

        <Category(Unary)>
        Public LogicalNot As Expression = IIFE(Function()
                                                   Dim b = True
                                                   Return Expr(Function() Not b)
                                               End Function)

        <Category(Unary)>
        Public TypeAs As Expression = IIFE(Function()
                                               Dim o As Object = Nothing
                                               Return Expr(Function() TryCast(o, String))
                                           End Function)
    End Module
End Namespace
